rules/app.bzl

```diff
@@ -56,7 +57,16 @@ def ios_application(name, apple_library = apple_library, **kwargs):
         apple_library: The macro used to package sources into a library.
         **kwargs: Arguments passed to the apple_library and ios_application rules as appropriate.
     """
-    infoplists = write_info_plists_if_needed(name = name, plists = kwargs.pop("infoplists", []))
+
+    infoplists_by_build_setting = kwargs.pop("infoplists_by_build_setting", {})
+    for (build_setting, plists) in infoplists_by_build_setting.items():
+        name_suffix = build_setting_name(build_setting)
+        infoplists_by_build_setting[build_setting] = write_info_plists_if_needed(name = "%s.%s" % (name, name_suffix), plists = plists)
+
+    infoplists_by_build_setting["//conditions:default"] = write_info_plists_if_needed(name = name, plists = kwargs.pop("infoplists", []))
```

Comment:
So when a target has both `infoplists_by_build_setting` and `infoplists`, `infoplists` will be used for default condition? Feels a little weird to put default in a separate attribute.

Reply:
Yep, by default the same value as before this change is being used and with the same name `write_info_plists_if_needed(name = name, ...)`, but for each build setting a file with a different name is being written above `write_info_plists_if_needed(name = "%s.%s" % (name, name_suffix), ...)`

If `infoplists_by_build_setting` is not empty it means that the consumer is saying "for the build setting `:blue` use this infoplists array instead of the one in the 'infoplists' attr"